src/codegen/encoding/mod.rs

```diff
@@ -1732,10 +1732,17 @@ pub(super) trait AbiEncoding {
 
     /// Returns if the we are packed encoding
     fn is_packed(&self) -> bool;
+
+    /// Encode constant data at compile time.
+    ///
+    /// Returns `None` if the data can not be encoded at compile time.
+    fn const_encode(&self, _args: &[Expression]) -> Option<Vec<u8>> {
+        None
+    }
```

Comment:
How about returning `Vec<u8>` and a default function which just calls `unimplemented!();`

Reply:
I like returning the Option way better. It allows to do `const_encode(..).unwrap_or(abi_encode())`

Reply:
I had not thought of that way